### PR TITLE
Increase the number of targets that are built for Bluepill.

### PR DESCRIPTION
**Pull request by advaitjain**
This will help provide some safeguard against issues like #43126

Note:
 * Flash and ram are much larger than the actual hardware.
 * The tests are only built, not run because we are currently unable to use renode as part of out continuous integration system.
 * Some of the core tests (such as micro_interpreter_test) do not build for the bluepill target and have been explicitly excluded.
 * While the kernel_circular_buffer_test builds for bluepill, it does not pass and so had to be excluded.

Manually tested that micro/tools/ci_build/test_bluepill.sh passes on my local machine (i.e. builds and all the tests that are run, pass).

### CODE REVIEW

File: tensorflow/lite/micro/tools/make/targets/bluepill/bluepill.lds [hunk 1]
```diff
@@ -36,8 +32,8 @@ ENTRY(_main)
 
 /* 20K of RAM and 128K of FLASH */
 MEMORY {
-RAM (xrw) : ORIGIN = 0x20000000, LENGTH = 20K
-FLASH (rx) : ORIGIN = 0x8000000, LENGTH = 128K
+RAM (xrw) : ORIGIN = 0x20000000, LENGTH = 256K
```
Comment (njeffrie) [inline tensorflow/lite/micro/tools/make/targets/bluepill/bluepill.lds:35]: Edit comment above?

File: tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc [hunk 2]
```diff
@@ -58,12 +54,24 @@ ifeq ($(TARGET), bluepill)
...
   MICROLITE_CC_SRCS := $(filter-out $(EXCLUDED_SRCS), $(MICROLITE_CC_SRCS))
   TEST_SCRIPT := tensorflow/lite/micro/testing/test_bluepill_binary.sh
-  # These are tests that don't currently work on the blue pill.
+
+  # TODO(b/143286954): Figure out why some tests fail and enable ince the issues
+  # are resolved.
   EXCLUDED_TESTS := \
     tensorflow/lite/micro/micro_interpreter_test.cc \
-    tensorflow/lite/micro/simple_tensor_allocator_test.cc
+    tensorflow/lite/micro/micro_allocator_test.cc \
```
Comment (njeffrie) [inline tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc:62]: I assume these tests never worked for bluepill?

**Review by njeffrie — DISMISSED** [review]

File: tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc (hunk 2 above)
Comment (advaitjain) [reply tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc:62]: not in the recent past. I have an internal bug to track this down.

File: tensorflow/lite/micro/tools/make/targets/bluepill/bluepill.lds (hunk 1 above)
Comment (advaitjain) [reply tensorflow/lite/micro/tools/make/targets/bluepill/bluepill.lds:35]: removed the unnecessary comment.

**Review by njeffrie — APPROVED** [review]

**Pull request merged**